WordPress/Classes/Login/WordPressDotComAuthenticator.swift

```diff
@@ -17,7 +18,64 @@ struct WordPressDotComAuthenticator {
         case unknown(Swift.Error)
     }
 
+    @MainActor
+    func signIn(from viewController: UINavigationController) async {
+        let token: String
+        do {
+            token = try await authenticate(from: viewController)
+        } catch {
+            if let error = error as? WordPressDotComAuthenticator.Error {
+                // Show an alert for non-cancellation errors.
+                // `.cancelled` error is thrown when user taps the cancel button in the presented Safari view controller.
+                if case .cancelled = error {
+                    // Do nothing
+                } else {
+                    // All other errors are unexpected.
+                    wpAssertionFailure("WP.com web login failed", userInfo: ["error": "\(error)"])
+
+                    let alert = UIAlertController(
+                        title: NSLocalizedString("wpComLogin.error.title", value: "Error", comment: "Error"),
+                        message: SharedStrings.Error.generic,
+                        preferredStyle: .alert
+                    )
+                    alert.addAction(UIAlertAction(title: SharedStrings.Button.close, style: .cancel, handler: nil))
+                    viewController.present(alert, animated: true)
+                }
+            } else {
+                wpAssertionFailure("WP.com web login failed", userInfo: ["error": "\(error)"])
+            }
+            return
```

Comment:
(nit) Extract the first part into a separate method so that there is no `return` in the middle of this long method.

Reply:
Addressed in https://github.com/wordpress-mobile/WordPress-iOS/pull/23675/commits/43409ded44f3491151584eb8d00a0bfdcfe76166